db/migrate/20180129034322_change_media_user_id_to_profile_id.rb

```diff
@@ -1,5 +1,20 @@
 class ChangeMediaUserIdToProfileId < ActiveRecord::Migration[5.1]
   def change
-    rename_column :media, :user_id, :profile_id
+    add_column :media, :profile_id, :bigint
+    media_type_to_role = {
+      "photo" => "Photographer",
+      "illustration" => "Illustrator",
+    }
+    Medium.find_each do |medium|
+      title = media_type_to_role[medium.media_type]
+      profile = Profile
+                  .joins(:role, :user)
+                  .find_by(
+                    "roles.title = ? AND users.id = ?",
+                    title,
+                    medium.user_id)
+      medium.update(profile_id: profile.id)
```

Comment:
If no `Profile` matches the condition, one should be created with the correct `user_id` and `role`. The `user_id` column isn't removed until after the loop anyways.